dask/array/reductions.py

```diff
@@ -320,7 +320,30 @@ def nanmax(a, axis=None, keepdims=False, split_every=None, out=None):
 
 def numel(x, **kwargs):
     """ A reduction to count the number of elements """
-    return chunk.sum(np.ones_like(x), **kwargs)
+
+    if hasattr(x, 'mask'):
+        return chunk.sum(np.ones_like(x), **kwargs)
+
+    shape = x.shape
+    keepdims = kwargs.get('keepdims', False)
+    axis = kwargs.get('axis', None)
+    dtype = kwargs.get('dtype', np.float64)
+
+    if axis is None:
+        prod = np.prod(shape, dtype=dtype)
+        return np.full((1,) * len(shape), prod, dtype=dtype) if keepdims is True else prod
+
+    if not isinstance(axis, tuple or list):
+        axis = [axis]
+    else:
+        axis = sorted(axis)
```

Comment:
Just curious, why does it matter if it is sorted?

Reply:
It doesn't. It might speedup search, but I now got rid of it in favor of readability.